clang/docs/checkedc/Update-to-latest-LLVM-sources.md

```diff
@@ -1,81 +1,251 @@
+
+Note that Phase 1 and the sub-phases of Phase 2 may not be executed one after
+another in one continuous stretch. They may be spread over several weeks and may
+be interspersed with Checked C development on the `master` branch.
+
+## Detailed Instructions - Phase 1
+
+1. Create a new branch off the `baseline` branch in the `checkedc-clang`
+repository. Let us call this new branch `updated_baseline`.
+```        
+    git checkout -b updated_baseline baseline
+```     
+2. Upgrade the branch `updated_baseline` to &lt;branch-point-of-12-on-main&gt;
+commit hash on the branch `main` of the LLVM/Clang repository. Record the commit
+hash &lt;branch-point-of-12-on-main&gt; in the document LLVM-Upgrade-Notes.md.
+```
+    git remote add upstream https://github.com/llvm/llvm-project.git 
+    git pull upstream <branch-point-of-12-on-main>
+```
+3. Execute  LLVM/Clang tests on the branch `updated_baseline` on your local
+machine to make sure that the upgrade is stable.
+
+4. Merge the `updated_baseline` branch back to the `baseline` branch. We expect
+branch `updated_baseline` to be a descendant of branch `baseline`.
+```
     git checkout baseline
-    git checkout -b updated-baseline
-    git pull mirror master
-
-## Run testing on your branched baseline branches.
-
-You can run testing locally or push your branched baseline branches to GitHub
-and use automated testing.  This will show you whether there are any unexpected
-failures in your baseline branch.
-
-If you use automated testing, make sure to clean the build directory first.
-Enough changes will likely have accumulated that things may go wrong without doing
-that first.
-
-## Branch your new baseline branches and merge master changes
-
-You can now branch your baseline branches to create a new master branch:
-
-    git checkout -b updated-master
+    git merge --ff-only updated_baseline
+```
+5. Create a new branch, say, `updated_baseline_master_12` from the `baseline`
+branch.
+```
+    git checkout -b updated_baseline_master_12 baseline
+```
+6. Merge the `master` branch of the checkedc-clang repository into the
+`updated_baseline_master_12` branch. This merge may cause several merge
+conflicts and test case failures. After the merge command, git may suggest to
+you to set merge.renamelimit to a specific value. It is a good idea to set it as
+suggested and re-run the merge. The last section in this document gives
+guidelines on fixing the merge conflicts and test failures.
+```
+    git checkout updated_baseline_master_12
+    git merge master > merge_conflict_list.txt
+``` 
+7. The resolution of test failures may require you to cherry-pick commits from
+the upstream LLVM/Clang repository. Record the cherry-picked commits in the
+document LLVM-Upgrade-Notes.md. Let &lt;commit-to-cherry-pick&gt; be one
+such commit hash on branch &lt;cherry-pick-branch&gt; of the LLVM/Clang
+repository. 
+```
+    git fetch upstream <cherry-pick-branch>
+    git cherry-pick <commit-to-cherry-pick>
```

Comment:
```suggestion
    git cherry-pick -x <commit-to-cherry-pick>
```
Recording `(cherry picked from commit X)` may be helpful.  I think your team has done this in other contexts.